src/shogun/kalman/kalman_filter.h

```diff
@@ -0,0 +1,52 @@
+#ifndef KALMAN_FILTER_H_
+#define KALMAN_FILTER_H_
+#include <shogun/mathematics/eigen3.h>
+
+using Eigen::MatrixXd;
+using Eigen::VectorXd;
+
+class KalmanFilter {
```

Comment:
plz observe how other methods are being implemented... they should inherit from SGObject etc etc... i suppose this is going to be a transformer? then should inherit from Transformer